src/app/upload/shared/upload.service.spec.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * upload.service.spec.ts
+ * Unit tests for service providing upload.
+ *
+   Copyright 2018 Sebastian Pustelnik
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import { TestBed, inject } from '@angular/core/testing';
+import { Http, BaseRequestOptions, Response, ResponseOptions } from '@angular/http';
+import { MockBackend, MockConnection } from '@angular/http/testing';
+import { UploadService } from "./upload.service";
+
+describe('UploadService', () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+          MockBackend,
+          BaseRequestOptions,
+          {
+            provide: Http,
+            useFactory: (backendInstance: MockBackend, defaultOptions: BaseRequestOptions) => {
+              return new Http(backendInstance, defaultOptions);
+            },
+            deps: [MockBackend, BaseRequestOptions]
+          },
+          UploadService
+      ]
+    });
+  });
+
+  it('should inject UploadService', inject([UploadService], (service: UploadService) => {
+    expect(service).toBeTruthy();
+  }));
+
```

Comment:
Test for a case without failure is missing. We had it in previous version, why did it get deleted? We'd just like to check, whether it passes without an error, right?